File: Lab_6/Codes/RGB_TO_Channel.py
import numpy as np
import matplotlib.pyplot as plt
from PIL import Image

# Read the RGB image
rgb_image = Image.open('flower.jpg').convert('RGB')
rgb_array = np.array(rgb_image)

# Extract the Red, Green, and Blue channels
RedChannel = rgb_array[:, :, 0]
GreenChannel = rgb_array[:, :, 1]
BlueChannel = rgb_array[:, :, 2]

# Display the original RGB image and the individual channels
plt.figure(figsize=(10, 8))

# Display original RGB image
plt.subplot(2, 2, 1)
plt.imshow(rgb_image)
plt.title('Original RGB Image')
plt.axis('off')

# Display Red channel
plt.subplot(2, 2, 2)
plt.imshow(RedChannel, cmap='Reds')
plt.title('Red Channel')
plt.axis('off')

# Display Green channel
plt.subplot(2, 2, 3)
plt.imshow(GreenChannel, cmap='Greens')
plt.title('Green Channel')
plt.axis('off')

# Display Blue channel
plt.subplot(2, 2, 4)
plt.imshow(BlueChannel, cmap='Blues')
plt.title('Blue Channel')
plt.axis('off')

plt.tight_layout()
plt.show()

File: Lab_6/Codes/RGB_TO_Channel.m
% Read the RGB image
rgbImage = imread('flower.jpg'); % Replace with your image file

% Extract the Red, Green, and Blue channels
RedChannel = rgbImage(:,:,1);
GreenChannel = rgbImage(:,:,2);
BlueChannel = rgbImage(:,:,3);

% Display the original RGB image and the individual channels
figure('Position', [100, 100, 800, 600]); % Set figure size

% Display original RGB image
subplot(2, 2, 1);
imshow(rgbImage);
title('Original RGB Image');
axis off;

% Display Red channel
subplot(2, 2, 2);
imshow(RedChannel);
title('Red Channel');
axis off;

% Display Green channel
subplot(2, 2, 3);
imshow(GreenChannel);
title('Green Channel');
axis off;

% Display Blue channel
subplot(2, 2, 4);
imshow(BlueChannel);
title('Blue Channel');
axis off;

% Adjust layout
tight_layout();
